src/object/cli_obj.c

```diff
@@ -3302,7 +4070,7 @@ obj_list_common(tse_task_t *task, int opc, daos_obj_list_t *args)
 
 	if ((args->dkey_anchor != NULL &&
 	     daos_anchor_get_flags(args->dkey_anchor) & DIOF_TO_LEADER) ||
-	    (obj_auxi->is_ec_obj && !obj_auxi->io_retry))
+	     obj_auxi->is_ec_obj)
```

Comment:
not sure why originally with !obj_auxi->io_retry?

Reply:
I suppose it means enum is only sent to leader for normal case. but I move all of checking case below to make it clear. so I remove this io_retry check.